src/panels/developer-tools/service/developer-tools-service.ts

```diff
@@ -353,19 +376,50 @@ class HaPanelDevService extends LitElement {
 
   private async _callService(ev) {
     const button = ev.currentTarget as HaProgressButton;
-    if (!this._serviceData?.service) {
+
+    if (this._yamlMode && !this._yamlValid) {
+      forwardHaptic("failure");
+      button.actionError();
+      showToast(this, {
+        message: this.hass.localize(
+          "ui.panel.developer-tools.tabs.services.errors.yaml.invalid_yaml"
+        ),
+      });
       return;
     }
-    const [domain, service] = this._serviceData.service.split(".", 2);
+
+    const { target, fields } = this._fields(
+      this.hass.services,
+      this._serviceData?.service
+    );
+
+    const isValid = this._isValid(
+      this._serviceData,
+      fields,
+      target,
+      this._yamlMode,
+      this.hass.localize
+    );
+
+    if (!isValid) {
+      forwardHaptic("failure");
+      button.actionError();
+      showToast(this, { message: this._error! });
```

Comment:
I would not use a toast here, but show an `ha-alert` like you did before and like we do on most other pages. @matthiasdebaat ?

Reply:
Can we remove all use of toasts on this page? I used it because errors from core on executing the service were previously displayed as toasts here, shall we convert all types of errors to ha-alert? 

I didn't personally like them either but just following what was previously there.

Reply:
Yes, I agree with that.

Reply:
If we display the alert on clicking call-service, when is the correct time to dismiss it? On any change to the yaml/ui, or wait until the next time call service is clicked?

I was leaving it up until call service button is re-clicked, though it feels a little wrong that the error instructs someone to select a target, and then after they select a target, the error is still prominently displayed.

Reply:
It's nice to dismiss the alert when it's fixed, but this also creates the expectation that it's going the be shown again when creating errors. To make it easy and predictable, I would go for clicking on the `call service` button.

Reply:
Ok, latest update moves all error types to an ha-alert (as in the images in the OP) and removes the toast.

Displayed error is cleared on: 
 - pressing call service that runs successfully
 - swtiching to a different service
 - toggling yaml mode (as error message is somewhat tailored to the mode)